### PR TITLE
policy: Add wildcard tests with and without L4

### PR DESCRIPTION
**Pull request by tgraf**

### CODE REVIEW

**Comment by tgraf** [comment]
test-me-please

**Comment by tgraf** [comment]
test-me-please

File: pkg/policy/rule_test.go [hunk 1]
```diff
@@ -1107,3 +1110,193 @@ func (ds *PolicyTestSuite) TestL4RuleLabels(c *C) {
...
+
+	endpointSelectorC = api.NewESFromLabels(labels.ParseSelectLabel("id=c"))
+
+	ctxAToB = SearchContext{From: labelsA, To: labelsB, Trace: TRACE_VERBOSE}
+	ctxAToC = SearchContext{From: labelsA, To: labelsC, Trace: TRACE_VERBOSE}
+)
+
+func expectResult(c *C, expected, obtained api.Decision, buffer *bytes.Buffer) {
+	if obtained != expected {
+		c.Errorf("Unexpected result: obtaiend=%v, expected=%v", obtained, expected)
```
Comment (nebril) [inline pkg/policy/rule_test.go]: nitpick: s/obtaiend/obtained

**Review by nebril — APPROVED** [review]
One typo. LGTM

**Comment by tgraf** [comment]
Noticed that tests are still incomplete

**Comment by ianvernon** [comment]
@tgraf let me know if you want me to pick up any egress work with this.

**Comment by tgraf** [comment]
> @tgraf let me know if you want me to pick up any egress work with this.

Just finished validating this

File: pkg/policy/rule_test.go (hunk 1 above)
Comment (tgraf) [reply pkg/policy/rule_test.go]: fixed

**Comment by tgraf** [comment]
test-me-please

**Comment by tgraf** [comment]
test-me-please

File: pkg/policy/rule_test.go [hunk 2]
```diff
@@ -1143,3 +1146,228 @@ func (ds *PolicyTestSuite) TestL4RuleLabels(c *C) {
...
+	endpointSelectorC = api.NewESFromLabels(labels.ParseSelectLabel("id=c"))
+
+	ctxAToB = SearchContext{From: labelsA, To: labelsB, Trace: TRACE_VERBOSE}
+	ctxAToC = SearchContext{From: labelsA, To: labelsC, Trace: TRACE_VERBOSE}
+)
+
+func expectResult(c *C, expected, obtained api.Decision, buffer *bytes.Buffer) {
+	if obtained != expected {
+		c.Errorf("Unexpected result: obtained=%v, expected=%v", obtained, expected)
+		c.Log(buffer)
```
Comment (ianvernon) [inline pkg/policy/rule_test.go:1175]: This is a great addition for logging policy trace!

**Review by ianvernon — APPROVED** [review]

**Pull request merged**